accepted/2025/PrunePackageReference-with-direct-PackageReference.md

```diff
@@ -0,0 +1,264 @@
+Note that this potential runtime issue is a general problem with packages that multi-target, where the dependencies of less specific frameworks like netstandard2.0 cannot be broader than more the specific frameworks like net9.0.  
+
+In the same, a reconsideration of the pruning behavior for direct package references was also suggested.
+
+## Explanation
+
+This design builds on the previous pruning specs, [PrunePackageReference roll-out](../2025/prune-package-reference-rollout.md) and [PrunePackageReference](../2024/prune-package-reference.md).
+
+### Scenarios for direct PackageReference items
+
+The motivation for the pruning overall is that we'd want to reduce the overhead of customers managing components that can be managed by the runtime.
+We want customers to remove unnecessary PackageReference items.
+As a long as a package is specified within a project but it's not needed, it adds maintenance overhead:
+
+- It may be accidentally updated by tooling such as dependabot
+- It may appear as a direct reference in the Package Manager UI or list package or others tools that list the project dependencies.
+- It is still part of auditing functionality such as NuGetAudit, deprecation etc.
+- It will be listed as a dependency of packages (thus needing them to be pruned.)
+
+Given that pruning is a per framework feature, there are a few scenarios to consider for the handling of direct package references, primarily revolving around multi-targeting:
+
+1. Single framework
+If the direct package reference is within the pruning range, we ask the customer to remove the PackageReference and reduce the management overhead.
+
+2. Multi framework scenario
+a. .NET & .NET Framework scenario
+
+```csproj
+  <PropertyGroup>
+    <TargetFramework>net9.0;net472</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="9.0.4" />
+  </ItemGroup>
+```
+
+For .NET, the package is unnecessary, and ideally we'd want the customer to consider conditionally reference the System.Text.Json package.
+Note that even in these situations, there's a theoretical way to end up with runtime issues, but it would require that a .NET Framework package references a package that references both, and then a .NET application reference this package.
+
+b. .NET & .NET Standard scenario
+
+```csproj
+  <PropertyGroup>
+    <TargetFramework>net9.0;netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="9.0.4" />
+  </ItemGroup>
+```
+
+For .NET, the package is unnecessary, but if the customer removes the reference and then at a later point point updates the PackageReference without bumping the target framework, it may lead to a situation where the netstandard2.0 version of their package has a broader API surface area than net9.0 and potentially leading to runtime issues as talked about in the [Motivation](#motivation).
```

Comment:
For `net9.0` the package is unnecessary.
```suggestion
For .NET 9, the package is unnecessary, but if the customer removes the reference and then at a later point point updates the PackageReference without bumping the target framework, it may lead to a situation where the netstandard2.0 version of their package has a broader API surface area than net9.0 and potentially leading to runtime issues as talked about in the [Motivation](#motivation).
```